lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/HttpTrackingURLProvider.java

```diff
@@ -47,7 +48,11 @@
    */
   @Inject
   public HttpTrackingURLProvider(final HttpServer httpServer) {
-    this.httpServer = httpServer;
+    if (httpServer instanceof DefaultHttpServerImpl) {
+      this.httpServer = null;
+    } else {
+      this.httpServer = httpServer;
+    }
```

Comment:
Maybe, write in one line, e.g.

    this.httpServer = httpServer instanceof NoOpHttpServerImpl ? null : httpServer;

(matter of taste, actually)